src/codeu/chat/util/Time.java

```diff
@@ -66,4 +66,10 @@ public String toString() {
 
   public static Time now() { return Time.fromMs(System.currentTimeMillis()); }
 
+  // Difference of end and start times.
+  // Ratio is the power of ten from seconds that the duration will be given in
+  // (e.i. ratio = -3 is milliseconds)
+  public static long duration(Time start, Time end, int ratio) {
+    return Math.round(Math.pow(10, -(3 + ratio)) * ((end.inMs() - start.inMs())));
```

Comment:
I mean that if we only keep one function (i.e. duration function) at the end of the day when we compare two durations, we don't have any information about the units. For example, if I find a duration which happens to be 3 seconds, that's going to be less (if compared) to a duration that is 300 milliseconds, simply because 3 is less that 300 (which is incorrect). If we make an immutable class (shouldn't take too long to do), we could keep the ratio as an instance variable so that when we do comparisons, we can take the ratio into consideration. Right now we are not using that but it may be helpful in the future.